rust/lance-encoding/src/encoder.rs

```diff
@@ -154,13 +155,15 @@ pub trait FieldEncoder: Send {
 
 pub struct BatchEncoder {
     pub field_encoders: Vec<Box<dyn FieldEncoder>>,
+    pub field_id_to_column_index: Vec<(i32, i32)>,
 }
 
 impl BatchEncoder {
     pub(crate) fn get_encoder_for_field(
-        field: &Field,
+        field: &LanceField,
```

Comment:
nit: in other places we use Lance `Field` and do `arrow_schema::Field as ArrowField`.